nni/compression/pytorch/speedup/jit_translate.py

```diff
@@ -461,179 +162,311 @@ def forward(self, obj):
-    c_node = node.key_node
-    inputs = list(c_node.inputs())
-    size_list_node = inputs[1].node()
-    scale_list_node = inputs[3].node()
-    size_list = None
-    scale_list = None
-
-    if size_list_node.kind() == 'prim::ListConstruct':
-        size_list = translate_list(inputs[1], speedup)
-    if scale_list_node.kind() == 'prim::ListConstruct':
-        scale_list = translate_list(inputs[3], speedup)
-    return UpsampleModule(size_list, scale_list)
-
-
-def upsample_nearest2d_python(node, speedup):
-    class UpsampleModule(torch.nn.Module):
-        def __init__(self, size_list, scale_list):
-            super(UpsampleModule, self).__init__()
-            self.size_list = size_list
-            self.scale_list = scale_list
-
-        def forward(self, *args):
-            """
-            The first input of args is the target tensor to upsample
-            , the following parameters is useless, because we already
-            get the size_list and the scale_list by parsing the cpp_nodes.
-            """
-            return torch.nn.functional.upsample_nearest(args[0],
-                                                        size=self.size_list, scale_factor=self.scale_list)
-    c_node = node.key_node
-    inputs = list(c_node.inputs())
-    size_list_node = inputs[1].node()
-    scale_list_node = inputs[2].node()
-    size_list = None
-    scale_list = None
+    """
+
+    func: Callable
+    positional: List[Any]
+    keyword: Dict[str, Any]
+    undetermined: List[Union[int, str]]
+    special_treat: Dict[Union[int, str], Callable]
+
+    def __new__(cls, func: Callable, positional: List[Any], keyword: Dict[str, Any],
+                undetermined: List[Union[int, str]], special_treat: Dict[Union[int, str], Callable]):
+        if not callable(func):
+            raise TypeError("the 'func' argument must be callable")
+
+        self = super(func_adapter, cls).__new__(cls)
+
+        self.func = func
+        self.positional = positional
+        self.keyword = keyword
+        self.undetermined = undetermined
+        self.special_treat = special_treat
+        return self
+
+    def __call__(self, /, *args):
+        assert len(args) >= len(self.undetermined)
+        if len(args) > len(self.undetermined):
+            logger.warning('throw some args away when calling the function \'%s\'', self.func.__name__)
+        for i, p in enumerate(self.undetermined):
+            v = args[i]
+            if type(p) is int:
+                self.positional[p] = v
+            else:
+                self.keyword[p] = v
+
+        for p, fs in self.special_treat.items():
+            if isinstance(p, int):
+                for f in fs: self.positional[p] = f(self.positional[p])
+            else:
+                for f in fs: self.keyword[p] = f(self.keyword[p])
+        result = self.func(*self.positional, **self.keyword)
+        if type(result) is int: # turn result of 'size' into tensor
```

Comment:
why need turn `int` into `Tensor`?

Reply:
the legacy code assumes that the output of each func op is a tensor or a tuple or list of tensors. there is only the int should be transformed in the origin code, and i did't find other types to be transformed. i'll try to do refactor in the next pr.